src/client/components/modals/CreatePackageModal.tsx

```diff
@@ -22,14 +22,18 @@ const CreatePackageModal: React.FC<CreatePackageModalProps> = ({ isOpen, setOpen
   const [cardName, setCardName] = useState<string>('');
   const [packageName, setPackageName] = useState<string>('');
   const [imageDict, setImageDict] = useState<{ [key: string]: { id: string } }>({});
+  const [fetched, setFetched] = useState(false);
 
   useEffect(() => {
-    fetch('/cube/api/imagedict')
-      .then((response) => response.json())
-      .then((json) => {
-        setImageDict(json.dict);
-      });
-  }, []);
+    if (isOpen && !fetched) {
```

Comment:
Here we could check if imageDict is empty or not instead of having explicit fetched. I implemented this first for the loading of followers and since a cube/user may have no followers, I needed an explicit state to check if loaded as I could not depend on data size. So to be consistent used the same fetched state in these modals.